File: src/main/java/fi/quanfoxes/types/Tiny.java
package fi.quanfoxes.types;

import fi.quanfoxes.lexer.NumberType;

public class Tiny extends Number {
    private static final int BYTES = 1;
    
    public Tiny() {
        super(NumberType.INT8, 8, "tiny");
    }

    @Override
    public int getSize() {
        return BYTES;
    }
}

File: Zigzag/Types/Tiny.cs
public class Tiny : Number
{
	private const int BYTES = 1;

	public Tiny() : base(Format.INT8, 8, false, "tiny") 
	{ 
		Identifier = "c";
	}

	public override void AddDefinition(Mangle mangle)
	{
		mangle.Value += "c";
	}

	public override int GetReferenceSize()
	{
		return BYTES;
	}

	public override int GetContentSize()
	{
		return BYTES;
	}
}
